packages/node_modules/nav-frontend-ekspanderbartpanel/src/ekspanderbartpanel-base-pure.tsx

```diff
@@ -62,26 +64,42 @@ class EkspanderbartpanelBasePure extends React.Component<EkspanderbartpanelBaseP
             onClick,
             collapseProps,
             border,
+            renderContentWhenClosed,
             ...renderProps
         }: EkspanderbartpanelBasePureProps = this.props;
 
+        const renderHiddenContent = renderContentWhenClosed && !apen;
+        const hiddenContentId = guid();
         return (
-            <div className={cls(apen, border!, className)} { ...renderProps }>
+            <div className={cls(apen, border!, className)} {...renderProps}>
                 <button
                     className="ekspanderbartPanel__hode"
                     onKeyDown={(event) => this.tabHandler(event)}
                     onClick={onClick}
                     aria-expanded={apen}
+                    aria-controls={renderHiddenContent ? hiddenContentId : undefined}
+                    role={renderHiddenContent ? 'tab' : undefined}
                     type="button"
                 >
                     <div className="ekspanderbartPanel__flex-wrapper">
                         {heading}
                         <span className="ekspanderbartPanel__indikator" />
                     </div>
                 </button>
-                <Collapse isOpened={apen} onRest={this.onRestProxy} {...collapseProps} >
-                    <article aria-label={ariaTittel} className="ekspanderbartPanel__innhold">{children}</article>
+                <Collapse
+                    isOpened={apen}
+                    onRest={this.onRestProxy}
+                    {...collapseProps}
+                >
+                    <article aria-label={ariaTittel} className="ekspanderbartPanel__innhold">
+                        {children}
+                    </article>
                 </Collapse>
+                {renderHiddenContent &&
+                    <div role="tabpanel" className="ekspanderbartPanel__hiddenContent" id={hiddenContentId}>
```

Comment:
Ref. første kommentar så kan `role="tabpanel"` fjernes her. Bør sikkert legge inn `aria-label={ariaTittel}` her også.

Reply:
Jeg hadde bare en teori om at google muligens vektet innholdet opp dersom det var en tab + tabpanel som var knyttet sammen, det er da tydeligere at det ikke er informasjon som er aktivt i siden, og ikke informasjon som kun er lagt inn for å få bedre rating. F.eks. når løsninger hadde mye innhold på siden som var skjult, bare for å lure google.

Reply:
Jau, skjønner 👍 Tror nok vi burde underbygge slike grep med noe konkret dokumentasjon hvis det skal være aktuelt. Uansett litt uheldig om komponenten leses ulikt av skjermlesere om man bruker `renderHiddenContent` eller ikke.

Har litt tro på at Google sine algoritmer klarer å se om informasjonen vi skjuler er relevant og meningsfylt eller bare et triks for å lure dem. Men det kan sikkert være greit å bruke `aria-controls` gjennomgående for å gjøre det implisitt at det skjulte innholdet kan toggles av brukeren ved behov.

Reply:
Har fjerner role, og setter nå id på collapse. Dersom collapseProps.id er satt, brukes denne som hiddenContentId, dersom ikke genereres en med guid();

Reply:
Hm, topp, men har undersøkt litt mer, og det ser ut som at dette kan løses på en litt enklere måte. Virker som at `react-collapse` eksponerer [to versjoner](https://github.com/nkbt/react-collapse/blob/4.x/src/index.js) av komponenten: `Collapse` og `UnmountClosed`, hvor [vi bruker sistnevnte](https://github.com/navikt/nav-frontend-moduler/blob/master/packages/node_modules/nav-frontend-ekspanderbartpanel/src/ekspanderbartpanel-base-pure.tsx#L3) som default i `EkspanderbartPanel` (og som navnet tilsier er det versjonen hvor `children` unmountes når panelet lukkes, mens den andre ikke gjør det).

Tenker vi kanskje burde se litt nærmere på dette i stedet? 🙂

Reply:
Ja, det er nok bedre. Jeg sjekker ut

Reply:
Kan det være en risiko for at applikasjoner vil feile dersom vi alltid bruker vanlig Collapse som rendrer content? Muligens noen som har basert seg på at innholdet rendres kun gitt at visse verdier eksisterer, og dersom en nå da alltid rendrer vil en risikere å prøve å rendre ting som er undefined -> fail

Reply:
Har pushet en versjon nå som gjør at den fungerer som tidligere, så sant en ikke har satt renderContentWhenClosed til true. Bruker deretter UnmountCollapse eller Collapse ut fra samme verdi.

Reply:
Hm, tenker vi bør sikre bakoverkompatibilitet ved å fortsette å bruke `UnmountClosed` som default,  men at vi lar konsumenten bruke `Collapse` hvis de ønsker. Prop'en som skrur på dette kan sikkert fortsette å hete `renderHiddenContent` eller noe slikt.

Reply:
Flott 👍 Så ikke siste kommentaren din før jeg postet forrige.

Reply:
Finnes det noe dokumentasjon som sier hvordan man får Googles crawler til å lese innhold, men ikke skjermlesere? Holder det med å legge på `[aria-hidden="true"`?

Reply:
@thomasfl Google sin crawler leser så vidt jeg vet alt innhold på siden, inkludert ting som er skjult med CSS eller andre grep. Skjermlesere ignorerer som regel ting som er skjult med CSS, men man kan også eksplisitt skjule synlig innhold ved å bruke [`aria-hidden="true"`](https://www.w3.org/TR/wai-aria-1.1/#aria-hidden).